test/unit/uk/gov/hmrc/tariffclassificationfrontend/controllers/CancelRulingControllerSpec.scala

```diff
@@ -103,16 +103,30 @@ class CancelRulingControllerSpec extends WordSpec with Matchers with UnitSpec
 
     "return OK and HTML content type" in {
       when(casesService.getOne(refEq("reference"))(any[HeaderCarrier])).thenReturn(successful(Some(caseWithStatusCOMPLETED)))
-      when(casesService.cancelRuling(refEq(caseWithStatusCOMPLETED), refEq(operator), any[Clock])(any[HeaderCarrier])).thenReturn(successful(caseWithStatusCANCELLED))
+      when(casesService.cancelRuling(refEq(caseWithStatusCOMPLETED), refEq(CancelReason.ANNULLED), refEq(operator), any[Clock])
+      (any[HeaderCarrier])).thenReturn(successful(caseWithStatusCANCELLED))
 
-      val result: Result = await(controller.confirmCancelRuling("reference")(newFakePOSTRequestWithCSRF(app)))
+      val result: Result = await(controller.confirmCancelRuling("reference")(newFakePOSTRequestWithCSRF(app).withFormUrlEncodedBody("reason" -> "ANNULLED")))
 
       status(result) shouldBe Status.OK
       contentTypeOf(result) shouldBe Some(MimeTypes.HTML)
       charsetOf(result) shouldBe Some("utf-8")
       bodyOf(result) should include("This case has been cancelled")
     }
 
+    "display required field when failing to submit reason" in {
```

Comment:
I'd maybe suggest this kind of test goes in the ViewSpec for that component e.g.` InputRadioGroupViewSpec`